tests/test_select.py

```diff
@@ -85,7 +86,7 @@ def test_select_default(
         pytest.param({"missing": "???"}, "missing", id="missing"),
     ],
 )
-def test_select_default_throw_on_missing(
+def test_select_default_throw_on_missing_dict(
```

Comment:
old function name seems better. why rename it?

Reply:
Before there was just the `test_select_default_throw_on_missing` function;
now there is a `test_select_default_throw_on_missing_dict` function and a `test_select_default_throw_on_missing_list` function.

Reply:
this function is testing both dicts and lists.

Reply:
It was only testing a dict here, but actually you're right that I can re-use the same test for lists as well (somehow I thought that `set_struct` was only for DictConfig, but looks like I was wrong).

I changed it in cef0a23